XcodeServerSDK/XcodeServerConfig.swift

```diff
@@ -33,16 +42,51 @@ public class XcodeServerConfig : JSONSerializable {
         return dict
     }
     
-    public init(var host: String, user: String?, password: String?) {
+    ///
+    /// Initializes a server configuration with the provided host.
+    /// - parameter host: `Xcode` server host.
+    /// - paramater user: Username that will be used to authenticate against the `host` provided.
+    /// Can be `nil`.
+    ///
+    /// - parameter password: Password that will be used to authenticate against the `host` provided.
+    /// Can be `nil`.
+    ///
+    /// - returns: A fully initialized `XcodeServerConfig` instance.
+    ///
+    /// - throws:
+    ///     - `InvalidHostProvided`: When the host provided doesn't produce a valid `URL`
+    ///     - `InvalidSchemeProvided`: When the provided scheme is not `HTTPS`
+    ///
+    public required init(var host: String, user: String?, password: String?) throws {
+        guard let url = NSURL(string: host) else {
+            /*******************************************************************
```

Comment:
Yeah I saw that on Twitter. Do you want to try it, @esttorhe? It's not the prettiest of things, but using implicitly unwrapped optional parameters would make it possible to use the new Swift 2 error throwing paradigm much more feasible. And when they finally fix it we can just get rid of the `!`s.